### DISCUSSION TITLE
During the translation process, the output content has formatting errors.

### DISCUSSION THREAD

**Discussion by jwdg001** [machinewrapped/llm-subtrans#96] [General]
When translating from English to Chinese, it is required to only retain the translated Chinese content, but the original English content sentences are also generated into the translated Chinese content. In addition, the original 793 subtitle lines were reduced to 759 subtitle lines after translation, and a translated Chinese subtitle line was duplicated to generate the same two lines of content.
![无标题546457546](https://github.com/machinewrapped/gpt-subtrans/assets/130415051/c231fcf3-dfa2-45dd-b176-09ef40e397bd)
![无标题567865746](https://github.com/machinewrapped/gpt-subtrans/assets/130415051/14aec564-a67e-4a54-8c93-a39d9e808f5d)

**Comment by machinewrapped** [comment]
Hi, for the English text appearing in the output, please make sure you don't have the "include original text" option checked :)
![image](https://github.com/machinewrapped/gpt-subtrans/assets/10140676/769537b4-270a-4ee7-a42d-33470937b8e7)

Assuming that's not it, it's probably because the default instructions are written in English, which predisposes GPT to responding in English. Try replacing the prompt with a Chinese prompt.

![image](https://github.com/machinewrapped/gpt-subtrans/assets/10140676/a73a1cfc-90ff-4ac5-9cee-54809aeaa468)

GPT learns well from examples, so if English -> Chinese is the language pair you use most it's probably worth modifying instructions.txt to reflect that, e.g. replace the first example (Japanese to English) with an English to Chinese example. You can save a custom instructions file and select it as the default in the settings. If you get better results, upload it here and I can include it in future releases!

**Comment by jwdg001** [comment]
你好，我没有勾选“包含原始文本”的选项，请原谅我使用中文来回复，我正在按你的提供的方法再次尝试进行翻译，目前还是会出现翻译好的内容，输出时格式出现错误，
主要存在的问题是输出的内容出现格式上的错误，比如在同一个时间轴下，它会把翻译好的中文句子重复输出变成两句一模一样的字幕内容，
还有就是会丢失一部分翻译内容，比如原本有800条英文字幕，翻译后只剩下700条字幕了。
同时我也用过另一个翻译工具，（https://github.com/gnehs/subtitle-translator-electron）它不会出现内容格式上输出的错误，但是翻译的准确效果会稍差一些，好像这个工具是专门用来把其它语言翻译成中文的缘故吧。
非常感谢你的回复，我正在尝试不同的选项和指令，得到有更好的结果我会告诉你的，谢谢。

**Comment by machinewrapped** [comment]
也许可以尝试使用更小的批次，这有助于GPT保持正确的轨道。我最近注意到在翻译时字幕合并的问题更多了。将出现错误的批次进行分割并重新翻译通常可以解决问题。

圣诞节期间我应该有时间来做这个项目，所以我想尝试使用Gemini来看看它的效果如何。翻译应该是它的强项之一，但我们还得看看它遵循指令的能力有多好。

**Comment by jwdg001** [comment]
好的，谢谢你的回复，稍后我会尝试的。
谷歌 Gemini 刚刚试了一下翻译字幕的效果，语言太生硬，对语言的理解会有错误的反馈，比如刚刚有一个句子，意思是不仅仅只是用手的力量，他翻译成用手的力量。用的是[沉浸式翻译](https://github.com/immersive-translate/immersive-translate)的上的字幕翻译插件。
chatgpt配合沉浸式翻译的字幕翻译功能，翻译出来的字幕，对上下文的理解很好，但同时他会把上下文的字幕合并在一起，对不上字幕的时间轴，把好多条时间轴上的内容，合并到一条时间轴上。

**Comment by machinewrapped** [comment]
我认为合并字幕是GPT最大的问题，我觉得它有时需要在源文中结合几行来理解意思，然后在翻译时无法将它们分开。

使用较小批次效果更好 - 我昨天翻译了一个采访，大约60行的批次在翻译中被压缩到大约20行，但当我将其分成大约20行的3个批次时，它设法在翻译中保持每一行的独立性。

如果合并行可以使翻译更好，我愿意让GPT这样做，然后调整开始和结束时间，使它们保持同步。我尝试时没能让重新同步部分工作，但我想再试一次。

**Comment by jwdg001** [comment]
是的，这个比较困难，结合上下文可以使翻译更准确，确实也会让字幕句子和时间轴对不上。
